### PR TITLE
Parallel support

### PR DESCRIPTION
**Pull request by Algunenano**
Implements #300

### CODE REVIEW

File: test/extension/run_at_cartodb_schema.sql [hunk 1]
```diff
@@ -1,7 +0,0 @@
-SET SCHEMA 'cartodb';
```
Comment (rafatower) [inline test/extension/run_at_cartodb_schema.sql]: why is this file deleted? it is referenced from the tests:
```
Makefile:       bash test/extension/test.sh
```

Comment (rafatower) [reply test/extension/run_at_cartodb_schema.sql]: oh, I see they were moved to `load_sql_file_schema` calls

**Comment by rafatower** [comment]
Files to change:
- CONTRIBUTING.md 
- the version in the makefile
- the NEWS.md

@jgoizueta can lend you a hand for this

File: test/extension/run_at_cartodb_schema.sql (hunk 1 above)
Comment (Algunenano) [reply test/extension/run_at_cartodb_schema.sql]: Yes, since `PARALLEL` needs to be stripped for 9.5- I couldn't just load the file directly.

**Comment by Algunenano** [comment]
* _CONTRIBUTING.md_: will do
* _Makefile_: I'd wait for more testing (e.g. having Crankshaft tested) to make a release. There are too many already (that need to be packaged in) and we can use latest-dev/latest-next for now. Or is this a requirement to deploy any change in staging/production?
* _NEWs_: is already [updated](https://github.com/CartoDB/cartodb-postgresql/pull/312/files#diff-8312ad0561ef661716b48d09478362f3), do you need anything else?

**Comment by rafatower** [comment]
> Makefile: I'd wait for more testing (e.g. having Crankshaft tested) to make a release. There are too many already (that need to be packaged in) and we can use latest-dev/latest-next for now. Or is this a requirement to deploy any change in staging/production?

It is. Usually you want to test upgrades and downgrades. Users already got the extension, if you don't bump the version an upgrade will do nothing. You cannot drop the extension the moment they have anything dependent upon it (unless you use CASCADE, but you don't want it).

**Comment by Algunenano** [comment]
You should be able to test it with an upgrade to `next`. Am I missing something?

Old package installed.
```sql
test_windshaft_cartodb_user_1_db=# CREATE DATABASE upgrade_test;
CREATE DATABASE
test_windshaft_cartodb_user_1_db=# \c upgrade_test
You are now connected to database "upgrade_test" as user "postgres".
upgrade_test=# \dx
                 List of installed extensions
  Name   | Version |   Schema   |         Description          
---------+---------+------------+------------------------------
 plpgsql | 1.0     | pg_catalog | PL/pgSQL procedural language
(1 row)

upgrade_test=# CREATE EXTENSION cartodb CASCADE;
NOTICE:  installing required extension "plpythonu"
NOTICE:  installing required extension "postgis"
CREATE EXTENSION
upgrade_test=# \dx
                                      List of installed extensions
   Name    | Version  |   Schema   |                             Description                             
-----------+----------+------------+---------------------------------------------------------------------
 cartodb   | 0.19.2   | cartodb    | Turn a database into a cartodb user database.
 plpgsql   | 1.0      | pg_catalog | PL/pgSQL procedural language
 plpythonu | 1.0      | pg_catalog | PL/PythonU untrusted procedural language
 postgis   | 2.5.0dev | public     | PostGIS geometry, geography, and raster spatial types and functions
(4 rows)

upgrade_test=# \df+ CDB_EqualIntervalBins
                                                                                                                  List of functions
 Schema  |         Name          | Result data type |        Argument data types        |  Type  | Volatility | Parallel |  Owner   | Security | Access privileges | Language |                      Source code                       | Description 
---------+-----------------------+------------------+-----------------------------------+--------+------------+----------+----------+----------+-------------------+----------+--------------------------------------------------------+-------------
 cartodb | cdb_equalintervalbins | anyarray         | in_array anyarray, breaks integer | normal | immutable  | unsafe   | postgres | invoker  |                   | sql      |                                                       +| 
         |                       |                  |                                   |        |            |          |          |          |                   |          | WITH stats AS (                                       +| 
         |                       |                  |                                   |        |            |          |          |          |                   |          |   SELECT min(e), (max(e)-min(e))/breaks AS del        +| 
         |                       |                  |                                   |        |            |          |          |          |                   |          |     FROM (SELECT unnest(in_array) e) AS p)            +| 
         |                       |                  |                                   |        |            |          |          |          |                   |          | SELECT array_agg(bins)                                +| 
         |                       |                  |                                   |        |            |          |          |          |                   |          |   FROM (                                              +| 
         |                       |                  |                                   |        |            |          |          |          |                   |          |     SELECT min + generate_series(1,breaks)*del AS bins+| 
         |                       |                  |                                   |        |            |          |          |          |                   |          |       FROM stats) q;                                  +| 
         |                       |                  |                                   |        |            |          |          |          |                   |          |                                                        | 
(1 row)
```
Here I installed the new package in my system.

```
upgrade_test=# ALTER EXTENSION cartodb UPDATE TO "0.19.2next";
ALTER EXTENSION
upgrade_test=# \df+ CDB_EqualIntervalBins
                                                                                                                  List of functions
 Schema  |         Name          | Result data type |        Argument data types        |  Type  | Volatility | Parallel |  Owner   | Security | Access privileges | Language |                      Source code                       | Description 
---------+-----------------------+------------------+-----------------------------------+--------+------------+----------+----------+----------+-------------------+----------+--------------------------------------------------------+-------------
 cartodb | cdb_equalintervalbins | anyarray         | in_array anyarray, breaks integer | normal | immutable  | safe     | postgres | invoker  |                   | sql      |                                                       +| 
         |                       |                  |                                   |        |            |          |          |          |                   |          | WITH stats AS (                                       +| 
         |                       |                  |                                   |        |            |          |          |          |                   |          |   SELECT min(e), (max(e)-min(e))/breaks AS del        +| 
         |                       |                  |                                   |        |            |          |          |          |                   |          |     FROM (SELECT unnest(in_array) e) AS p)            +| 
         |                       |                  |                                   |        |            |          |          |          |                   |          | SELECT array_agg(bins)                                +| 
         |                       |                  |                                   |        |            |          |          |          |                   |          |   FROM (                                              +| 
         |                       |                  |                                   |        |            |          |          |          |                   |          |     SELECT min + generate_series(1,breaks)*del AS bins+| 
         |                       |                  |                                   |        |            |          |          |          |                   |          |       FROM stats) q;                                  +| 
         |                       |                  |                                   |        |            |          |          |          |                   |          |                                                        | 
(1 row)
```

With the update `cdb_equalintervalbins`is now parallel safe (used to be unsafe).


Downgrade:
```sql
upgrade_test=# ALTER EXTENSION cartodb UPDATE TO "0.19.2";
ALTER EXTENSION
upgrade_test=# \df+ CDB_EqualIntervalBins
                                                                                                                  List of functions
 Schema  |         Name          | Result data type |        Argument data types        |  Type  | Volatility | Parallel |  Owner   | Security | Access privileges | Language |                      Source code                       | Description 
---------+-----------------------+------------------+-----------------------------------+--------+------------+----------+----------+----------+-------------------+----------+--------------------------------------------------------+-------------
 cartodb | cdb_equalintervalbins | anyarray         | in_array anyarray, breaks integer | normal | immutable  | safe     | postgres | invoker  |                   | sql      |                                                       +| 
         |                       |                  |                                   |        |            |          |          |          |                   |          | WITH stats AS (                                       +| 
         |                       |                  |                                   |        |            |          |          |          |                   |          |   SELECT min(e), (max(e)-min(e))/breaks AS del        +| 
         |                       |                  |                                   |        |            |          |          |          |                   |          |     FROM (SELECT unnest(in_array) e) AS p)            +| 
         |                       |                  |                                   |        |            |          |          |          |                   |          | SELECT array_agg(bins)                                +| 
         |                       |                  |                                   |        |            |          |          |          |                   |          |   FROM (                                              +| 
         |                       |                  |                                   |        |            |          |          |          |                   |          |     SELECT min + generate_series(1,breaks)*del AS bins+| 
         |                       |                  |                                   |        |            |          |          |          |                   |          |       FROM stats) q;                                  +| 
         |                       |                  |                                   |        |            |          |          |          |                   |          |                                                        | 
(1 row)
```

All the upgrade/dowgrade paths are exactly the same (same contents) so there is no way to go back to a non-parallel function, but that's nothing new, isn't it?

**Comment by rafatower** [comment]
You are not missing anything, but "upgrade next" is is the hacky way. For staging and production you want to version, tag, release and then deploy.

Unless you see a risk there, I'd go ahead, as we don't want to pile up changes if not strictly needed. If you see significant risk, then better think about possible mitigations.

IMHO it is better to merge and release (and eventually have to mark something as parallel unsafe) than to wait for some other piece to be ready.

**Comment by Algunenano** [comment]
I think those two last commits should address the required changes. If you have time, can you have a look at it again @rafatower ?
